framework/doc/content/syntax/FunctorMaterials/index.md

```diff
@@ -0,0 +1,169 @@
+# FunctorMaterials
+
+Functor material properties are properties that are evaluated on-the-fly.
+Along with numerous classes of MOOSE objects, they are [Functors](syntax/Functors/index.md). This allows
+for the creation of objects with considerable inter-operability between systems.
+
+## Using functor materials
+
+Much like regular materials declare regular material properties, functor materials declare functor material properties.
+Following a producer/consumer model, `FunctorMaterials` produce functor material properties, which kernels, boundary conditions, other
+functor materials, etc, may consume.
+Both regular and functor material properties support a wide array of data types (scalar, vector, tensors,
+[automatic differentiation](automatic_differentiation/index.md) versions of all of these). Functor materials, because they are evaluated on-the-fly
+and call other functors on-the-fly as well, do not require dependency ordering.
```

Comment:
Yes, another very important benefit which I often forget to bring up. Thanks for highlighting this!